src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1654,7 +1654,7 @@ trait Namers extends MethodSynthesis {
         if (sym hasAnnotation NativeAttr)
           sym resetFlag DEFERRED
         else {
-          if (!symbolAllowsDeferred && ownerRequiresConcrete) fail(AbstractVar)
+          if (!symbolAllowsDeferred && ownerRequiresConcrete && !sym.isJava) fail(AbstractVar)
```

Comment:
[no worries, I guess everyone was extremely busy with the latest milestone release]

I totally agree with your reasoning; that's also why I was quite puzzled as to why the java doc comments weren't showing up after my initial changes to the parser. Nevertheless, after some more searching, I found a method `def createJavadoc: Boolean = false` in `Global`, which appears to control if the typer is run on java.
`Global#GlobalPhase`:
```scala
def cancelled(unit: CompilationUnit) = {
  // run the typer only if in `createJavadoc` mode
  val maxJavaPhase = if (createJavadoc) currentRun.typerPhase.id else currentRun.namerPhase.id
  reporter.cancelled || unit.isJava && this.id > maxJavaPhase
}
```
It was added by Donna Malayeri in 2010, however I was unable to find any use of it in the current code base. My thinking was that `createJavadoc` must have been lost in refactoring over time (I did not do an extensive search with git though, since scalac has changed quite a bit since then and I'm only just getting familiar with the code base). Overriding it in `ScaladocGlobal` enters the typer and causes the errors that require the changes to namer and typer.